src/Result.php

```diff
@@ -242,4 +243,52 @@ private function ensureHasKeyValue(): void
             throw NoKeyValue::fromColumnCount($columnCount);
         }
     }
+
+    /**
+     * BC layer for a wide-spread use-case of old DBAL APIs
+     *
+     * @deprecated This API is deprecated and will be removed after 2022
+     *
+     * @return array<array-key,mixed>|false
+     */
+    public function fetch(int $mode = FetchMode::ASSOCIATIVE)
```

Comment:
As far as I understand, we can afford to have a type declaration here because the extending classes that don't provide a type declaration won't violate LSP but the same is not true about the return types. Right?

Reply:
My thought process was, that this is a new API so we can add whatever type declarations, however the return value is really `mixed` so i didn't declare it.

Reply:
In this case, should return type declarations be added everywhere else where possible?

Reply:
in DBAL in general or what do you mean?